src/test/java/com/snowplowanalytics/snowplow/tracker/emitter/BatchEmitterTest.java

```diff
@@ -355,6 +355,145 @@ public void noRetryAfterDenylistResponseCode() throws InterruptedException {
         Assert.assertEquals(0, emitter.getBuffer().size());
     }
 
+    @Test
+    public void callsSuccessCallbackAfterSending() throws InterruptedException {
+        class TestCallback implements EmitterCallback {
+            List<TrackerPayload> payloads;
+
+            @Override
+            public void onSuccess(List<TrackerPayload> payloads) {
+                this.payloads = payloads;
+            }
+
+            @Override
+            public void onFailure(FailureType failureType, boolean willRetry, List<TrackerPayload> payloads) {
```

Comment:
Perhaps we should throw an error in this case? Similar in the tests below for the cases that are not expected.

Reply:
What would the error do?

Reply:
I just wanted it to fail the test in case the wrong callback is called – to be sure that only the `onSuccess` callback is called and not the `onFailure` (or vice versa in other test cases). Otherwise you could have an implementation that calls both callbacks and passes the tests. (btw, forgot to say but this is really is just a nitpick)